src/cinnamon-util.c

```diff
@@ -540,7 +543,12 @@ cinnamon_util_format_date (const char *format,
 int
 cinnamon_util_get_week_start ()
 {
-  int week_start;
+  /* Try to get first weekday from gsettings */
+  /* If 7, continue to get the locale's first weekday */
+  GSettings *settings = g_settings_new (DESKTOP_SCHEMA);
+  int week_start = g_settings_get_int (settings, FIRST_WEEKDAY_KEY);
+  if (week_start != 7) return week_start;
```

Comment:
Changing this to

    if (0 <= week_start && week_start <= 7) return week_start;

should solve the problem